tide/master/rest/RestWindows.cpp

```diff
@@ -79,53 +117,65 @@ std::string RestWindows::_toJSON() const
             continue;
 
-       QJsonObject window;
-       ContentWindowController controller( *contentWindow , _displayGroup );
-       window["aspectRatio"] = contentWindow->getContent()->getAspectRatio();
-       window["minWidth"] = controller.getMinSizeAspectRatioCorrect().width();
-       window["minHeight"] = controller.getMinSizeAspectRatioCorrect().height();
-       window["width"] = contentWindow->getDisplayCoordinates().width();
-       window["height"] = contentWindow->getDisplayCoordinates().height();
-       window["x"] = contentWindow->getDisplayCoordinates().x();
-       window["y"] = contentWindow->getDisplayCoordinates().y();
-       window["z"] = _displayGroup.getZindex( contentWindow );
-       window["title"] = contentWindow->getContent()->getTitle();
-       window["mode"] = contentWindow->getMode();
-       window["selected"] = contentWindow->isSelected();
-       window["fullscreen"] = contentWindow->isFullscreen();
-       window["focus"] = contentWindow->isFocused();
-       window["uri"] = contentWindow->getContent()->getURI();
-       window["uuid"] = contentWindow->getID().toString().replace( _regex,"" );
-       arr.append(window);
+	QJsonObject window;
+        ContentWindowController controller( *contentWindow , _displayGroup );
+        window["aspectRatio"] = contentWindow->getContent()->getAspectRatio();
+        window["minWidth"] = controller.getMinSizeAspectRatioCorrect().width();
+        window["minHeight"] = controller.getMinSizeAspectRatioCorrect().height();
+        window["width"] = contentWindow->getDisplayCoordinates().width();
+        window["height"] = contentWindow->getDisplayCoordinates().height();
+        window["x"] = contentWindow->getDisplayCoordinates().x();
+        window["y"] = contentWindow->getDisplayCoordinates().y();
+        window["z"] = _displayGroup.getZindex( contentWindow );
+        window["title"] = contentWindow->getContent()->getTitle();
+        window["mode"] = contentWindow->getMode();
+        window["selected"] = contentWindow->isSelected();
+        window["fullscreen"] = contentWindow->isFullscreen();
+        window["focus"] = contentWindow->isFocused();
+        window["uri"] = contentWindow->getContent()->getURI();
+        window["uuid"] = _getUuid( *contentWindow );
+        arr.append(window);
     }
     QJsonObject obj;
     obj["windows"] = arr;
     return QJsonDocument{ obj }.toJson().toStdString();
 }
 
-QString _getThumbnailUri( const ContentWindow& window )
-{
-    const QString id = window.getID().toString().replace( _regex, "" );
-    return "tide/windows/" + id + "/thumbnail";
-}
-
-void RestWindows::_registerThumbnailUrl( ContentWindowPtr window )
+void RestWindows::_cacheThumbnail( ContentWindowPtr window )
 {
-    const auto url = _getThumbnailUri( *window );
-    const QString uri = window->getContent().get()->getURI();
-    _httpServer.handleGET(url.toStdString(), [uri] ()
+    QtConcurrent::run( QThreadPool::globalInstance(),
```

Comment:
I think that QThreadPool::globalInstance() is the default, it can be omitted (also remove the include)